modules/network/nat-gateway/main.bicep

```diff
@@ -0,0 +1,60 @@
+@description('Required. The name of the NAT Gateway resource.')
+param name string
+
+@description('Required. Location(region) for NAT Gateway will be deployed.')
+param location string
+
+@description('Optional. Tags for natGateways resource.')
+param tags object = {}
+
+@minValue(4)
+@maxValue(120)
+@description('Optional. The idle timeout of the NAT Gateway.')
+param idleTimeoutInMinutes int = 4
+
+@description('Optional. An array of public ip addresses associated with the nat gateway resource.')
+param publicIpAddresses array = []
+/*Example:
+publicIpAddresses: [{
+        id: '/subscriptions/XXX-XXX-XXX-XXX/resourceGroups/nat-test3/providers/Microsoft.Network/publicIPAddresses/nat-public-ip1-jgqel5bzokle6'
+      }]
+*/
+
+@description('Optional. An array of public ip prefixes associated with the nat gateway resource.')
+param publicIpPrefixes array = []
+/*Example:
+publicIPPrefixes: [{
+        id: '/subscriptions/XXX-XXX-XXX-XXX/resourceGroups/nat-test3/providers/Microsoft.Network/publicIPPrefixes/nat-public-ip-prefix-jgqel5bzokle6'
+      }]
+*/
+
+@description('Optional. Specify Azure Availability Zone IDs or leave unset to disable.')
+param zones array = []
```

Comment:
this toggle name is becoming common across services. I like to go with this to conform. And then we can give the zones a default value, which will make it easier for the users in most cases when they are using ZR since they wont have to worry about the array values. Only in the most advanced cases would they need to do something like [ '1' '2']